File: 알고리즘 스터디 문제들/Part 3 - 완전 탐색/추가 문제/한수.py
def han(N):
    flag = True
    N_str = str(N)
    for i in range(len(N_str) - 2):
        if (int(N_str[i + 1]) - int(N_str[i]) != int(N_str[i + 2]) - int(N_str[i + 1])):
            flag = False
    return flag


N = int(input())
cnt = 0
for i in range(1, N + 1):
    if (han(i)):
        cnt += 1
print(cnt)

File: 알고리즘 스터디 문제들/Part 3 - 완전 탐색/추가 문제/한수.cpp
#include <iostream>
#include <string>
using namespace std;

bool han(int N)
{
    bool flag = true;
    string str = to_string(N);
    for (size_t i = 2; i < str.length(); i++)
    {
        if (str[i] - str[i - 1] != str[i - 1] - str[i - 2])
            flag = false;
    }
    return flag;
}

int main()
{
    int N;
    cin >> N;

    int cnt = 0;
    for (size_t i = 1; i <= N; i++)
    {
        if (han(i))
            cnt++;
    }
    cout << cnt;
}
